src/contexts/ChainIdContext.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { createContext, useCallback, useState } from 'react';
+
+interface IChainIdContext {
+  chainId: number | null;
+  setChainId: (chainId: number) => void;
+}
+
+interface IProps {
+  children: React.ReactNode;
+}
+
+const chainIdStorageKey = 'chainId';
+const defaultChainId = _retrieveChainId();
+
+export const ChainIdContext = createContext<IChainIdContext>({
+  chainId: defaultChainId,
+  setChainId: () => {},
+});
+
+export const ChainIdProvider: React.FC<IProps> = ({ children }: IProps) => {
+  const [chainId, setChainId] = useState<number | null>(defaultChainId);
+
+  const handleSetChainId = useCallback((chainId: number | null) => {
+    _storeChainId(chainId);
+    setChainId(chainId);
+  }, []);
+
+  return (
+    <ChainIdContext.Provider value={{ chainId, setChainId: handleSetChainId }}>{children}</ChainIdContext.Provider>
+  );
+};
+
+function _retrieveChainId(): number | null {
+  const chainId = JSON.parse(localStorage.getItem(chainIdStorageKey) || 'null');
```

Comment:
localstrage.getitem wont return a string? On the web it says, 
Return Value: | A String, representing the value of the specified key

Reply:
That's why typescript is so great - the actual return value is a string OR `null` if there's nothing in storage. In that case `JSON.parse` would fail, hence the fallback to string `'null'` which can be parsed just fine